src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp

```diff
@@ -0,0 +1,417 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ProfileViewModel.h"
+#include "ProfileViewModel.g.cpp"
+#include "EnumEntry.h"
+
+#include <LibraryResources.h>
+#include "..\WinRTUtils\inc\Utils.h"
+
+// This function is a copy of DxFontInfo::_NearbyCollection() with
+// * the call to DxFontInfo::s_GetNearbyFonts() inlined
+// * checkForUpdates for GetSystemFontCollection() set to true
+static wil::com_ptr<IDWriteFontCollection1> NearbyCollection(IDWriteFactory* dwriteFactory)
+{
+    // The convenience interfaces for loading fonts from files
+    // are only available on Windows 10+.
+    wil::com_ptr<IDWriteFactory6> factory6;
+    // wil's query() facilities don't work inside WinRT land at the moment.
+    // They produce a compilation error due to IUnknown and winrt::Windows::Foundation::IUnknown being ambiguous.

```

Comment:
Could be a good EIM thing to see if this is already solved in an updated wil and/or contribute a fix to their side for it.

Reply:
Still doesn't work :(